File: Applications/SpaceWeather/SW1D_sqrt/mesh1D_adapted.py
from numpy import *
from scipy.optimize import fsolve

def mesh1D_adapted(r1,r2,nx):
    dlay = 1
    dwall = 1e-2

    #Calculate mesh ratio
    c = 1-dlay/dwall
    def function(x):
        return scalingfun(x,nx,c)

    rat = fsolve(function,1.1)
    rat = rat[0]

    xv = zeros(nx+1)
    xv[1] = dwall
    for i in range(1,nx):
        xv[i+1] = xv[i] + dwall*rat**i

    #if abs(xv[-1]-dlay)>1e-8:
        #error("Something wrong with the input parameters")

    p = array([xv*(r2-r1)+r1])
    t = array([[k,k+1] for k in range(0,nx)])
    t = t.transpose()
    return p,t


def scalingfun(x,n,c):
    F = c

    for i in range(1,n):
        F += x**i 

    return F

File: Applications/SpaceWeather/oldSingleSpecies/SW1D_singleSpecies_scaled/mesh1D_adapted.m
function [p,t] = mesh1D_adapted(r1,r2)

dlay = 1;
dwall = 1e-2; 
nx = 16;

% calculate the mesh ratio
c = 1 - dlay/dwall;
rat = fsolve(@(x) scalingfun(x,nx,c),[1;1.5]);
rat = rat(1);

% scaling distribution over the normal direction
xv = zeros(nx+1,1);
xv(2) = dwall;
for i = 1:(nx-1)
    xv(i+2) = xv(i+1) + dwall*(rat^i);
end

if abs(xv(end)-dlay)>1e-8
    error("Something wrong with the input parameters (dlay, dwall, nx)");
end

p = xv'*(r2-r1) + r1;
t = [(1:nx); (2:nx+1)];
